### PR TITLE
Refactor CoordinateSystem and Composite to handle their own scope caches.

### PR DESCRIPTION
**Pull request by wwwtyro**
What do you think about this, @flekschas? Will it solve most of your memory leak issue from #34? It's similar to your PR, just narrower in _scope_ (heh, heh):

- Uses your approach of passing the CG object into the coordinate system, which I think makes a lot of sense.
- Creates the the coordinate system scope at coordinate system construction time and stores it on the object. 
- Removes the coordinate scope cache from the CG object.
- Creates the composite scope at composite construction time and stores it on the object.
- Removes the composite scope cache from the CG object.

If this works for you, I'll take a stab at the `commandCache` cleanup separately.

(Side note: I still need to go update the API docs)

### CODE REVIEW

**Comment by flekschas** [comment]
Thanks for the PR! I'll try to test the it today

File: src/candygraph.ts [hunk 1]
```diff
@@ -89,19 +122,24 @@ export class CandyGraph {
...
-      );
-    });
-  };
+  public destroy() {
+    this.clearPositionBuffers();
+    this.regl.destroy();
+  }
+
+  private getCommand(coords: CoordinateSystem, primitive: Primitive) {
+    let m0 = this.commandCache.get(coords.glsl);
```
Comment (flekschas) [inline src/candygraph.ts]: Not critical at all: I wonder what `m0` stands for. How about calling it `commands`? To me `commands` carries more meaning. :)

**Review by flekschas — APPROVED** [review]
Thanks again for the PR. It definitely alleviates the memory issue but I wouldn't consider it solved because of the command cache and the inability to clear it.

One minor thing which would be nice to add is https://github.com/wwwtyro/candygraph/pull/36#discussion_r873055457. It could technically be it's own PR but you might as well squeeze it in here :) (If you think it's appropriate of course.)

**Comment by wwwtyro** [comment]
Thanks for the review! Agreed on your points, I'll get your recommended changes in. Should have this in sometime this evening.

**Comment by flekschas** [comment]
Thanks a lot for the PR 🎉

**Pull request merged**